packages/marshal/src/rankOrder.js

```diff
@@ -225,14 +225,18 @@ export const makeComparatorKit = (compareRemotables = (_x, _y) => 0) => {
   };
 
   /** @type {RankCompare} */
-  const antiComparator = (x, y) => comparator(y, x);
+  const outerComparator = (x, y) =>
+    /** @type {Exclude<PartialComparison, NaN>} */ (comparator(x, y) || 0);
```

Comment:
✅ narrowing the range in case of a compareRemotables that returns NaN, as it now will by default.

This does break any existing program that expects to distinguish NaN and provided a compareRemotables to produce that effect, but as you state you’re convinced nobody does that in practice, I’m fine with this. It may be worth noting in NEWS.md if only to build trust.

Reply:
It might be worth explicitly reminding reviewers that `NaN || 0 === 0` because it’s not entirely obvious which JavaScript operators implicitly coerce down from the float to integer domain.